Tradução para inglês (#6)
stack-spot/mind-dotnet-api-stack: docs/use-case_en-us.md

## About
The API Stack is non-intrusive. Templates and plugins are created to help development teams, but each developer chooses what they prefer to use.
By launching your product using the **base template** provided as an example, you can compile and run the project locally. 

The **dotnet-api-template** template adds to a Stack the feature that provisioning `REST` services.

> To start your development journey using the resources provided by **Mind Studio**, use this base template as an example to start developing a complete API using **C#**, **.NET** and **REST **, and running in a container cluster.  

Reviewer: Acompanhar use-case-observability

## Prerequisites
To use the **dotnet-api-template** template:
- Install [**`STK CLI`**](https://stackspot.com/download)
- **.NET 6**.

### **Inputs**
The inputs needed to use the template are: 

| **Field** | **Value** | **Description** |
| :--- | :--- | :--- |
| Target Framework| .net6 | Target Framework in which the application will be created  |

## Running the project

After creating the project, access the directory where it was created and run the following command:

- Replace the `*` with the given name for application.

```bash
dotnet restore *.Api.sln
```

Also build the project, using the command below:

```bash
dotnet build *.Api.sln
```

In order to run the unit and integration tests use the command below:

```bash
dotnet test *.Api.sln
```

To run the application, in the same directory, use the following command:

```bash
dotnet run --project ./src/*.Api/*.Api.csproj
```

### Docker Configurations

> In order for the application to work with Docker, you will need to add a temporary SSL certificate and mount a volume to hold that certificate.
> You can find the details in [Microsoft Docs](https://docs.microsoft.com/en-us/aspnet/core/security/docker-https?view=aspnetcore-6.0) that describe the necessary steps for Windows, macOS and Linux.

Reviewer 2: ```suggestion
If you want your application to work with Docker, you will need to add a temporary SSL certificate and mount a volume to hold that certificate.
For more details, see the [Microsoft Docs](https://docs.microsoft.com/en-us/aspnet/core/security/docker-https?view=aspnetcore-6.0) that describe the necessary steps for Windows, macOS, and Linux.
```

For Windows:

```bash
dotnet dev-certs https -ep %USERPROFILE%\.aspnet\https\aspnetapp.pfx -p Your_password123
dotnet dev-certs https --trust
```

> NOTE: When using PowerShell, replace **%USERPROFILE% with $env:USERPROFILE**.
For macOS:
```bash
dotnet dev-certs https -ep ${HOME}/.aspnet/https/aspnetapp.pfx -p Your_password123
dotnet dev-certs https --trust
```

For Linux:
```bash
dotnet dev-certs https -ep ${HOME}/.aspnet/https/aspnetapp.pfx -p Your_password123
dotnet dev-certs https --trust
```

> To build and run docker containers, run the command below in the solution root where you find the docker-compose.yml file.

Reviewer 2: ```suggestion
To build and run docker containers, go to the solution root where you find the **docker-compose.yml file** and execute the command below:
```

 ```bash
 docker-compose -f 'docker-compose.yml' up --build
 ```

> You can also use a **IDE** (VSCode, Visual Studio) of your choice to perform the above steps.

> With the application running, follow the steps below:  
> 1. Access the url https://localhost:5001/swagger;
> 2. If the browser prompts you about the site's security, click "Go to localhost (not secure)". By accessing the above address, you will see the documentation for your application.

Reviewer 2: ```suggestion
Your application is running, now:
1. Access https://localhost:5001/swagger to see more details.
2. If your browser prompts you about the website's security, click on **"Go to localhost (not secure)"**.  When you access this URL, you will see your application's documentation.
```

### Base Template Capabilities
Check below the capabilities available to be added to a base Template.

- **Logging**  

> If you have several projects, microservices and/or routines, with this Plugin you can maintain the quality and standard of the log information for everything, considering the different log levels.
> 
> It's allow you to define Tags so that you can quantify and/or classify errors, providing a view of recurring failures or preventing future problems.
> 
> In addition, it is easy to integrate with tools such as **Splunk** or **AWS Cloud Watch** to monitor and visualize data, ensuring the observability of your applications.

Reviewer 2: ```suggestion
If you have several projects, microservices, and/or routines, with this Plugin you can maintain the quality and standard of the log information for everything, considering the different log levels.
It allows you to define Tags so you can quantify and/or classify errors, providing a view of recurring failures or preventing future problems.
It is easy to integrate with tools like **Splunk** or **AWS Cloud Watch** to monitor and visualize data, ensuring the observability of your applications.
```

- **Metrics**    

> The Metrics Plugin makes it easy to measure the number of requests on a critical endpoint or monitor the size of payloads that are sent to your API.

Reviewer 2: ```suggestion
The Metrics Plugin makes it easy to measure the number of requests on a critical endpoint or monitor the size of payloads that are sent to your API.

```

- **Trace**    

> Trace Plugin facilitates telemetry instrumentation of your API, through **Opentelemetry**, with exporter for **Jaeger** and **AWS X-Ray**.

Reviewer: Acompanhar use-case-observability

> Trace Plugin makes it easy the telemetry instrumentation of your API, through **Opentelemetry**, with exporter for **Jaeger** and **AWS X-Ray**.

Reviewer 2: ```suggestion
Trace Plugin facilitates the telemetry instrumentation of your API, through **Opentelemetry**, with an exporter for **Jaeger** and **AWS X-Ray**.

```

- **Queue**    

> The Queue Plugin simplifies the entire configuration and implementation of a decentralized environment, focused on microservices that communicate with each other, using message queues. Compared to creating a class, Plugin Queue manages to abstract all the complexity for communicating with your **AWS SQS** queue.

Reviewer 2: ```suggestion
The Queue Plugin simplifies the entire configuration and implementation of a decentralized environment, focused on microservices that communicate with each other, using message queues. It's like creating a class, the Queue Plugin manages to abstract all the complexity for communicating with your **AWS SQS** queue.

```

- **Notification**    

> The Notification Plugin provides the features of **AWS SNS (Simple Notification Service)** so that your application can send messages to topics that will be consumed by your Subscribers, simultaneously sending data, information and messages to different platforms at once.

Reviewer 2: ```suggestion
The Notification Plugin provides the **AWS SNS (Simple Notification Service)** features so your application can send messages to topics that will be consumed by your Subscribers, simultaneously sending data, information, and messages to different platforms at once.
```

- **Secrets**    

> Plugin Secrets turns easier the management of sensitive system information with API keys, authentication tokens or database connections. It was thought of using a **Secret Manager** from AWS simplifying the consumption of this information.
> Use the Secrets plugin any time you want within your application.

Reviewer 2: ```suggestion
Secrets Plugin makes the management of sensitive system information with API keys, authentication tokens, or database connections easier. It was thought of using a **Secret Manager** from AWS to simplify the consumption of this information.
Use the Secrets plugin any time you want in your application.
```

- **Bucket**    

> Handle files using Plugin Bucket, it eliminates complexity and facilitates instrumentation with **Amazon Simple Storage Service (S3)**.

Reviewer 2: ```suggestion
It handles files using the Bucket Plugin, eliminating complexity and facilitating instrumentation with **Amazon Simple Storage Service (S3)**.
```

- **Repository**    

> Store data or documents trivially. Plugin Repository brings together all the capabilities of **AWS DynamoDB** in a simple and fast way.

Reviewer 2: ```suggestion
Store data or documents trivially. Repository Plugin brings together all the capabilities of **AWS DynamoDB** in a simple and fast way.
```

- **Authentication**    

> Plugin Authentication provides more security for you to easily verify a user/API accesses to the restricted resources of your API.

Reviewer: Acompanhar use-case-observability
